src/rust/ide/src/view/project_name.rs

```diff
@@ -31,21 +72,26 @@ impl ProjectName {
     , project_controller : &controller::Project
     , fonts              : &mut font::Registry
     ) -> Self {
+        let scene                 = world.scene();
+        let network               = frp::Network::new();
         let logger                = Logger::sub(&logger,"ProjectName");
         let display_object        = display::object::Instance::new(&logger);
         let font                  = fonts.get_or_load_embedded_font("DejaVuSansMono").unwrap();
         let size                  = Vector2::new(600.0,100.0);
-        let base_color            = color::Rgba::new(1.0, 1.0, 1.0, 0.7);
+        let base_color            = DARK_GRAY_TEXT_COLOR;
         let text_size             = 16.0;
-        let text_field_properties = TextFieldProperties{base_color,font,size,text_size};
+        let text_field_properties = TextFieldProperties{base_color,font,size:size.clone(),text_size};
         let text_field            = TextField::new(&world,text_field_properties);
-        text_field.set_content(&project_controller.project_name());
-        let project_controller = project_controller.clone();
-        display_object.add_child(&text_field.display_object());
-        Self {logger,display_object,text_field,project_controller}.initialize()
+        let project_controller    = project_controller.clone();
+        let main_logger           = Logger::sub(&logger,"main_area");
+        let main_area             = component::ShapeView::<shape::Shape>::new(&main_logger,scene);
+        Self{main_area,network,logger,display_object,text_field,project_controller}.initialize()
     }
 
     fn initialize(self) -> Self {
+        self.text_field.set_content(&self.project_controller.project_name());
+        self.display_object.add_child(&self.text_field.display_object());
+        self.display_object.add_child(&self.main_area);
```

Comment:
never use `display_object` explicitly. `self.display_object. ...` is bad. The same `(&self.text_field.display_object());` All things like that should ook just as `self.add_child(&self.text_field);` always.

Reply:
@wdanilo oh, it works for `self` but not for `TextField`. `TextField`'s `DisplayObject' is inside a `RefCell`, so it can't implement `display::Object` to get its reference.